fast64_internal/oot/oot_actor.py

```diff
@@ -1,47 +1,401 @@
+				# If it's not a Switch Flag, change nothing to the name
+				if lenSwitch is not None:
+					if int(lenSwitch) > 1: name = f'{elemName} #{i}'
+					else: name = elemName
+				else: name = elemName
+
+				# Set the name to none to use the element's name instead
+				# Set the name to the element's name if it's a flag and its name is set
+				curName = elem.get('Name')
+				if elemName is None or (elTag == 'Flag' and curName is not None): name = curName
+
+				# Add the index to get the proper attribute
+				if elTag == 'Property' or (elTag == 'Flag' and elType == 'Switch') or elTag == 'Bool' or elTag == 'Enum':
+					field = elemField + f'{i}'
+				else:
+					field = elemField
+
+				attr = getattr(detailedProp, field, None)
+				tiedParam = elem.get('TiedParam')
+				actorType = getattr(detailedProp, key + '.type', None)
+				if name != 'None' and elem.tag == elTag and elType == elem.get('Type') and attr is not None:
+					if hasActorTiedParams(tiedParam, actorType) is True:
+						prop_split(box, detailedProp, field, name)
+					i += 1
+
+def drawOperatorBox(layout, obj, detailedProp, field, labelText, searchOp, enum):
+	searchOp.objName = obj
+	split = layout.split(factor=0.5)
+	split.label(text=labelText)
+	split.label(text=getEnumName(enum, getattr(detailedProp, field)))
+
+def editOOTActorProperties():
+	'''This function is used to edit the OOTActorProperties class before it's registered'''
+	propAnnotations = getattr(OOTActorProperties, '__annotations__', None)
+	if propAnnotations is None:
+		propAnnotations = {}
+		OOTActorProperties.__annotations__ = propAnnotations
+
+	# Collectible Drops List
+	itemDrops = [(elem.get('Value'), elem.get('Name'),
+					elem.get('Name')) for listNode in root for elem in listNode if listNode.tag == 'List'
+					and listNode.get('Name') == 'Collectibles']
+
+	# Generate the fields
+	for actorNode in root:
+		i_property = i_flag = i_bool = i_enum = 1
+		actorKey = actorNode.get('Key')
+		for elem in actorNode:
+			if elem.tag == 'Property':
+				genString(propAnnotations, actorKey, f'.props{i_property}', actorKey)
+				i_property += 1
+			elif elem.tag == 'Flag':
+				if elem.get('Type') == 'Chest':
+					genString(propAnnotations, actorKey, '.chestFlag', 'Chest Flag')
+				elif elem.get('Type') == 'Collectible':
+					genString(propAnnotations, actorKey, '.collectibleFlag', 'Collectible Flag')
+				elif elem.get('Type') == 'Switch':
+					genString(propAnnotations, actorKey, f'.switchFlag{i_flag}', 'Switch Flag')
+					i_flag += 1
+			elif elem.tag == 'Collectible':
+				genEnum(propAnnotations, actorKey, '.collectibleDrop', itemDrops, 'Collectible Drop')
+			elif elem.tag == 'Parameter':
+				actorTypeList = [(elem2.get('Params'), elem2.text, elem2.get('Params'))
+								for actorNode2 in root for elem2 in actorNode2
+								if actorNode2.get('Key') == actorKey and elem2.tag == 'Parameter']
+				genEnum(propAnnotations, actorKey, '.type', actorTypeList, 'Actor Type')
```

Comment:
so this and Collectible (and maybe others, unsure) assume only one of these can be used by actor? it's an okay assumption to make I guess but it should be enforced by checking (or even better allow several of these)

Reply:
I think this is the "added numbers to more props" part of the commit so I'm gonna resolve this, feel free to unresolve if I'm wrong